### PR TITLE
fix(FE/dashboard): Configure dashboard drawer UX fixes

### LINKED ISSUE #3413: DASHBOARD [FE]: Configure drawer should close on save and other UI improvements
## Bug description

The configure drawer doesn't automatically close on saving changes. 

Other issues:

- Header padding is not proper 
- New tag button size not consistent with added tags
- New tag input doesn't close on clicking outside

image

## Expected behavior
The configure dashboard drawer should close automatically on successfully saving

## How to reproduce

1. Go to Dashboard page
2. Click on Configure button
3. Click on Save

### PR DESCRIPTION
**Pull request by CKanishka**
Fixes #3413 

**Description**

- Added padding to the close button on the header
- Drawer to be auto-closed on save
- Tags alignment and close on blur fixed

**Changes Overview**

https://github.com/SigNoz/signoz/assets/30779692/0050e445-9523-4df9-ab55-de7278e5f9e7

### CODE REVIEW

File: frontend/src/store/reducers/dashboard.ts [hunk 1]
```diff
@@ -29,6 +31,8 @@ const InitialValue: InitialValueTypes = {
 	errorMessage: '',
 	isEditMode: false,
 	isAddWidget: false,
+	isConfigureDrawerVisible: false,
+	generalConfigSaving: false,
```
Comment (Rajat-Dabade) [inline frontend/src/store/reducers/dashboard.ts:35]:
```suggestion
	isGeneralConfigSaving: false,
```

File: frontend/src/types/reducer/dashboards.ts [hunk 2]
```diff
@@ -7,4 +7,6 @@ export default interface DashboardReducer {
 	errorMessage: string;
 	isEditMode: boolean;
 	isAddWidget: boolean;
+	isConfigureDrawerVisible: boolean;
+	generalConfigSaving: boolean;
```
Comment (Rajat-Dabade) [inline frontend/src/types/reducer/dashboards.ts:11]: Same here

**Review by Rajat-Dabade — CHANGES_REQUESTED** [review]
One small change, everything else LGTM 🚀

**Comment by CKanishka** [comment]
> One small change, everything else LGTM 🚀

Addressed comment

**Review by Rajat-Dabade — APPROVED** [review]
LGTM

**Pull request closed without merging**